providers/fcontext.rb

```diff
@@ -0,0 +1,50 @@
+# Support whyrun
+def whyrun_supported?
+  true
+end
+
+use_inline_resources
+
+# Create if doesn't exist, do not touch if fcontext is already registered
+action :add do
+  escaped_file_spec = Regexp.escape(new_resource.file_spec)
+  noregex_file_spec = new_resource.file_spec.split(/(\(\/\.\*\)\?|\.\*)$/)[0]
```

Comment:
We'll avoid having to expand it because we'll pass it to `find` (as the
regex), which will expand it by itself.
But I think I obsessed over it too much. Let's do whatever and if someone
complains we'll invest the time in fixing it. Please keep a `#XXX` above it
though :)

*Nitzan Raz*
http://backslasher.net

On Mon, Mar 2, 2015 at 3:59 PM, Jörg Herzinger <notifications@github.com>
wrote:

> In providers/fcontext.rb
> <https://github.com/BackSlasher/chef-selinuxpolicy/pull/3#discussion_r25599503>
> :
>
> > @@ -0,0 +1,50 @@
> > +# Support whyrun
> > +def whyrun_supported?
> > +  true
> > +end
> > +
> > +use_inline_resources
> > +
> > +# Create if doesn't exist, do not touch if fcontext is already registered
> > +action :add do
> > +  escaped_file_spec = Regexp.escape(new_resource.file_spec)
> > +  noregex_file_spec = new_resource.file_spec.split(/(\(\/\.\*\)\?|\.\*)$/)[0]
>
> 1: This is more of less pseudo-code. I somehow tested it, and it will
> work, but you know...
> How could you ever expand a .* without checking the actual FS?
> 2: If we'd use -R we could drop the find all along. ;)
>
> I think we have a plan. I'll try to get that ready within the next few
> hours.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/BackSlasher/chef-selinuxpolicy/pull/3/files#r25599503>
> .
>